Data Structures/BinaryTree/BT_DiaPair.java

```diff
@@ -0,0 +1,163 @@
+
+import java.util.Scanner;
+
+public class BT_DiaPair {
+
+	static Scanner scn = new Scanner(System.in);
+
+	public static void main(String[] args) {
+
+		BT_DiaPair bts = new BT_DiaPair();
+		BinaryTree bt = bts.new BinaryTree();
+		// 10 true 20 true 40 false false true 50 false false true 30 true 60 false
+		// false false **Give this Input to create the tree**
+
+		System.out.println(bt.DiaPair());
+
+	}
+
+	private class BinaryTree {
+
+		private class Node {
+			int data;
+			Node left;
+			Node right;
+		}
+
+		private Node root;
+		private int size;
+
+		public BinaryTree() {
+			this.root = construct(null, false);
+		}
+
+		private Node construct(Node parent, boolean ilc) {		// ilc -> This is for asking if left child is present then pass 										   true else pass false
+
+			if (parent == null) {
+				System.out.println("Enter the data for root node ? ");
+			} else {
+				if (ilc) {
+					System.out.println("Enter the data for left " + parent.data);
+				} else {
+					System.out.println("Enter the data for right " + parent.data);
+				}
+			}
+
+			int val = scn.nextInt();	// val -> for generating the node
+			Node nn = new Node();		// nn -> stand for new node
+			nn.data = val;
+			this.size++;
+
+			boolean lc;			// lc -> left child

```

Comment:
Do not substitute proper variable names with comments. If the variable is less than 20 chars there is no need to abbreviate it. It just makes it harder to read. Look [here.](http://wiki.c2.com/?GoodVariableNames)